runtime/doc/news.txt

```diff
@@ -30,6 +30,8 @@ The following changes may require adaptations in user config or plugins.
     set mousemodel=popup
     set keymodel=startsel,stopsel
 <
+• |CursorMoved| autocommand now triggers on the next main loop iteration rather
+  than immediately when switching windows. This is more compatible with Vim.
```

Comment:
Please offer a suggestion for users who are relying on the current behavior. Can they use a combination of `WinEnter` + `CursorMoved` to achieve the current behavior?

Reply:
> You only need to add `WinEnter` if you rely on the event firing **immediately** for active window changes. Is that your case?

I think it is. My autocommands have to do with setting up highlights and some other visual elements whenever the cursor position changes. I'm interested in

- Whenever `[line, col]` changes within the same window (`CursorMoved`)
- Whenever the cursor moves to another window, regardless of its `[line, col]` position in the new window. `CursorMoved` was useful for covering that as well, but I suppose I'll need to add `WinEnter` after this change.

Reply:
I don't think you'll need that. If `CursorMoved` was previously enough for you, it should still be enough for you after this change.

Reply:
When the cursor is moved to a different window, `CursorMoved` will only be triggered **after** I move the cursor within the new window, right? i.e. it will no longer be triggered as soon as I enter the window? The aforementioned "setting up highlights and some other visual elements" should happen as soon as I enter the window, hence why I think `WinEnter` will be necessary - because `CursorMoved` would no longer be useful for detecting changes in the active window.

Reply:
Are you pressing a key to move cursor, or are you doing that using a command or function that immediately follows the command that changes the active window? If you are pressing a key that `CursorMoved` is triggered before you will be able to press any keys.

Reply:
I think the documentation changes from the latest pushes helped in clarifying this point. Suppose that I have this autocommand:

```vim
autocmd CursorMoved * call SetupHighlights()
```

I want SetupHighlights to be called

1 - When I change the cursor's position within the current window
2 - When I move the cursor to a different window

After this PR, 2 will no longer occur **immediately**, only when Neovim resumes the main loop. The difference is, if I have:

```vim
function! Foo()
  call MoveCursorToAnotherWindow()
  call DoSomething()
endfunction
```

Upon calling Foo(), SetupHighlights() would not be triggered after MoveCursorToAnotherWindow() because `CursorMoved` is not triggered immediately. `CursorMoved` will only be fired at the end of Foo(), when Neovim would go back to the main loop. In other words, it's still possible to detect active window changes with `CursorMoved`, but the event will no longer be fired immediately. Is that correct? If it is, I suppose `WinEnter` is not needed for the highlights use-case.

The new text is clearer, but it has a typo. It says "back the main loop", but it should be "back to the main loop" or "back in the main loop".

Reply:
Yes